core/templates/dev/head/pages/exploration_editor/editor_tab/StateResponsesDirective.js

```diff
@@ -0,0 +1,519 @@
+                }
+              ]
+            }).result.then(function() {
+              var deletedOutcome =
+                ResponsesService.getAnswerGroup(index).outcome;
+              ResponsesService.deleteAnswerGroup(index);
+              var deletedFeedbackContentId =
+                deletedOutcome.feedback.getContentId();
+              stateContentIdsToAudioTranslationsService.
+                displayed.deleteContentId(deletedFeedbackContentId);
+              stateContentIdsToAudioTranslationsService.saveDisplayedValue();
+            });
+          };
+
+          $scope.saveActiveAnswerGroupFeedback = function(updatedOutcome) {
+            ResponsesService.updateActiveAnswerGroup({
+              feedback: updatedOutcome.feedback
+            });
+          };
+
+          $scope.saveActiveAnswerGroupDest = function(updatedOutcome) {
+            ResponsesService.updateActiveAnswerGroup({
+              dest: updatedOutcome.dest,
+              refresherExplorationId: updatedOutcome.refresherExplorationId,
+              missingPrerequisiteSkillId:
+                updatedOutcome.missingPrerequisiteSkillId
+            });
+          };
+
+          $scope.saveActiveAnswerGroupCorrectnessLabel = function(
+              updatedOutcome) {
+            ResponsesService.updateActiveAnswerGroup({
+              labelledAsCorrect: updatedOutcome.labelledAsCorrect
+            });
+          };
+
+          $scope.saveActiveAnswerGroupRules = function(updatedRules) {
+            ResponsesService.updateActiveAnswerGroup({
+              rules: updatedRules
+            });
+          };
+
+          $scope.saveDefaultOutcomeFeedback = function(updatedOutcome) {
+            ResponsesService.updateDefaultOutcome({
+              feedback: updatedOutcome.feedback
+            });
+          };
+
+          $scope.saveDefaultOutcomeDest = function(updatedOutcome) {
+            ResponsesService.updateDefaultOutcome({
+              dest: updatedOutcome.dest,
+              refresherExplorationId: updatedOutcome.refresherExplorationId,
+              missingPrerequisiteSkillId:
+                updatedOutcome.missingPrerequisiteSkillId
+            });
+          };
+
+          $scope.saveDefaultOutcomeCorrectnessLabel = function(updatedOutcome) {
+            ResponsesService.updateDefaultOutcome({
+              labelledAsCorrect: updatedOutcome.labelledAsCorrect
+            });
+          };
+
+          $scope.getAnswerChoices = function() {
+            return ResponsesService.getAnswerChoices();
+          };
+
+          $scope.isOutcomeLooping = function(outcome) {
+            var activeStateName = EditorStateService.getActiveStateName();
+            return outcome && (outcome.dest === activeStateName);
+          };
+
+          $scope.navigateToState = function(stateName) {
+            RouterService.navigateToMainTab(stateName);
+          };
+        }
+      ]
+    };
+  }]);
+
+oppia.filter('summarizeAnswerGroup', [
```

Comment:
It might be tidier to turn these into functions, or at least break out into separate files? Just noting this because usually we have a one-directive-per-file rule.

Reply:
Done, turned them into functions.